Detectors/FIT/FT0/simulation/src/Detector.cxx

```diff
@@ -870,7 +870,11 @@ void Detector::defineFrameTransformations()
 Bool_t Detector::ProcessHits(FairVolume* v)
 {
 
-  TString volname = fMC->CurrentVolName();
+  Int_t copy;
+  Int_t volID = fMC->CurrentVolID(copy);
+  Int_t volREG = fMC->VolId("0REG");
+  Int_t volTOP = fMC->VolId("0TOP");
+  Int_t volMTO = fMC->VolId("0MTO");
```

Comment:
these should be cached as data members inside the Detector class to avoid calling `VolId` again and again.